src/coreclr/jit/stacklevelsetter.cpp

```diff
@@ -89,16 +105,10 @@ void StackLevelSetter::ProcessBlock(BasicBlock* block)
             SubStackLevel(numSlots);
         }
 
-#if !FEATURE_FIXED_OUT_ARGS
-        // Set throw blocks incoming stack depth for x86.
-        if (throwHelperBlocksUsed && !framePointerRequired)
+        if (throwHelperBlocksUsed && node->OperMayThrow(comp))
```

Comment:
Since `GTF_CALL` and `GTF_EXCEPT` are up-to-date and overapproximations here you can optimize TP with this trick:
https://github.com/dotnet/runtime/blob/3805c174d0a72dadfdbef98011b10b32df9e93f3/src/coreclr/jit/promotion.cpp#L2340-L2349